app/shared/lib/triggerDownload.ts

```diff
@@ -0,0 +1,9 @@
+export const triggerDownload = async (data: string, filename: string, type?: string): Promise<void> => {
+    const blob = new Blob([Buffer.from(data, 'base64')], type ? { type } : {});
```

Comment:
I'm not familiar with the type of data being downloaded, but is it possible that the data or filename could be malicious? if so, we could update this method with some validations 
```suggestion
export const triggerDownload = async (data: string, filename: string, type?: string): Promise<void> => {
    // Validate base64 format
    if (!/^[A-Za-z0-9+/]*={0,2}$/.test(data)) {
        throw new Error('Invalid base64 data');
    }

    // Sanitize filename to prevent directory traversal
    const sanitizedFilename = filename.replace(/[^a-z0-9._-]/gi, '_');

    // Limit file size to prevent DoS (e.g., 10MB)
    const decoded = Buffer.from(data, 'base64');
    if (decoded.length > 10 * 1024 * 1024) {
        throw new Error('File too large to download');
    }

    const blob = new Blob([Buffer.from(data, 'base64')], type ? { type } : {});
```